### PR TITLE
*: lower-case-table-names should be removed from tidb-config

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/32885

Problem Summary:

lower-case-tables-names should not be in configuration.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
The configuration option `lower-case-table-names` has no effect in TiDB. It has been removed from configuration.
```

### CODE REVIEW

**Review by xiongjiwei — APPROVED** [review]

File: executor/brie.go [hunk 1]
```diff
@@ -285,9 +285,9 @@ func (b *executorBuilder) buildBRIE(s *ast.BRIEStmt, schema *expression.Schema)
 		cfg.TableFilter = filter.All()
 	}
 
-	if tidbCfg.LowerCaseTableNames != 0 {
-		cfg.TableFilter = filter.CaseInsensitive(cfg.TableFilter)
```
[the remark this answers is not in the document]
Comment (morgo) [reply executor/brie.go:289]: It looks like it is used to specify if the tablefilter should act case insensitive.  I just changed it to always be case insensitive, which looks like the correct behavior anyway.

[the remark this answers is not in the document]
Comment (3pointer) [reply executor/brie.go:289]: LGTM, I think it's okay to always be case insensitive. since we only support case insensitive in tidb.

**Review by bb7133 — APPROVED** [review]
LGTM

**Pull request merged**